cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -657,6 +657,13 @@ else()
            "${THIRDPARTY_MIRROR_URL}/snappy-${ARROW_SNAPPY_BUILD_VERSION}.tar.gz")
 endif()
 
+# Remove these two lines once https://github.com/substrait-io/substrait/pull/342 merges
```

Comment:
Moving forward sounds fine to me; this is why we're pulling this dependency from a URL after all. I've added https://issues.apache.org/jira/browse/ARROW-18201 to track the reversion. In the meantime, please move this modification to [versions.txt](https://github.com/westonpace/arrow/blob/cdb16492027dec59e6b8c55440077bd0dd3c919d/cpp/thirdparty/versions.txt#L84-L85), annotated with `// TODO(ARROW-18201)`